addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairBaseBridgeHandler.java

```diff
@@ -0,0 +1,470 @@
+                return t;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Class for throwing an End of Buffer exception, used in getByte when read returns a -1
+     *
+     * @author Jeff James - initial contribution
+     *
+     */
+    public class EOBException extends Exception {
+        private static final long serialVersionUID = 1L;
+    }
+
+    /**
+     * Gets a single byte from m_reader input stream
+     *
+     * @param s used during debug to identify proper state transitioning
+     * @return next byte from m_reader
+     * @throws EOBException
+     * @throws IOException
+     */
+    private int getByte(PARSERSTATE s) throws EOBException, IOException {
+        int c = 0;
+
+        try {
+            c = m_reader.read();
+            if (c == -1) {
+                throw new EOBException();
+            }
+        } catch (IOException e) {
+            throw e;
+        }
+        // logger.debug("BUF: " + byteToHex(c) + " " + s);
+        return c;
+    }
+
+    /**
+     * Gets a specific number of bytes from m_reader input stream
+     *
+     * @param buf byte buffer to store bytes
+     * @param start starting index to store bytes
+     * @param n number of bytes to read
+     * @return number of bytes read
+     * @throws EOBException
+     * @throws IOException
+     */
+    private int getBytes(byte[] buf, int start, int n) throws EOBException, IOException {
+        int i;
+        int c;
+
+        try {
+            for (i = 0; i < n; i++) {
+                c = m_reader.read();
+                if (c == -1) {
+                    throw new EOBException();
+                }
+
+                buf[start + i] = (byte) c;
+            }
+
+        } catch (IOException e) {
+            throw e;
+        }
+
+        return i;
+    }
+
+    /**
+     * Job to send pump query status packages to all Intelliflo Pump things in order to see the status.
+     * Note: From the internet is seems some FW versions of EasyTouch controllers send this automatically and this the
+     * pump status packets can just be snooped, however my controller version does not do this. No harm in sending.
+     *
+     * @author Jeff James
+     *
+     */
+    class PumpStatus implements Runnable {
+        @Override
+        public void run() {
+            // FF 00 FF A5 00 60 10 07 00 01 1C
+            List<Thing> things = getThing().getThings();
+            byte[] packet = { (byte) 0xA5, (byte) 0x00, (byte) 0x00, (byte) id, (byte) 0x07, (byte) 0x00 };
+
+            PentairPacket p = new PentairPacket(packet);
+
+            for (Thing t : things) {
+                if (!t.getThingTypeUID().equals(INTELLIFLO_THING_TYPE)) {
+                    continue;
+                }
+
+                p.setDest(((PentairIntelliFloHandler) t.getHandler()).m_id);
+                writePacket(p);
+                try {
+                    Thread.sleep(2000);
```

Comment:
I reduced the delay here.  Unfortunately, I don't have multiple pumps to test this.